packages/zoe/test/unitTests/contracts/test-escrowToVote.js

```diff
@@ -0,0 +1,217 @@
+    t.rejects(
+      () => E(voter).vote({ 'not a question': 'NO' }),
+      /The question 'not a question' did not match the question 'Should we upgrade\?'/,
+      `A vote for the wrong question throws`,
+    );
+    console.log('EXPECTED ERROR ->>>');
+    t.rejects(
+      () => E(voter).vote({ [QUESTION]: 'NOT A VALID ANSWER' }),
+      /the answer 'NOT A VALID ANSWER' was not 'YES' or 'NO'/,
+      `A vote with an invalid answer throws`,
+    );
+
+    const result1 = await E(voter).vote({ [QUESTION]: 'YES' });
+    t.equals(
+      result1,
+      `Successfully voted ${{ [QUESTION]: 'YES' }}`,
+      `voter2 votes YES`,
+    );
+
+    // Votes can be recast at any time
+    const result2 = await E(voter).vote({ [QUESTION]: 'NO' });
+    t.equals(
+      result2,
+      `Successfully voted ${{ [QUESTION]: 'NO' }}`,
+      `voter 2 recast vote for NO`,
+    );
+
+    payoutP.then(async payout => {
+      const moolaPayment = await payout.Assets;
+
+      t.deepEquals(
+        await moolaIssuer.getAmountOf(moolaPayment),
+        moola(5),
+        `voter2 gets everything she escrowed back`,
+      );
+
+      console.log('EXPECTED ERROR ->>>');
+      t.throws(
+        () => voter.vote({ [QUESTION]: 'NO' }),
+        /the escrowing offer is no longer active/,
+        `voter2 voting fails once offer is withdrawn or amounts are reallocated`,
+      );
+    });
+  };
+
+  await voter2Votes(voterInvite2);
+
+  // Voter 3 votes NO and then completes their offer, meaning that
+  // their vote should not be counted. They get their full moola (1
+  // moola) back as a payout.
+  const voter3Votes = async invite => {
+    const proposal = harden({
+      give: { Assets: moola(1) },
+    });
+    const payments = harden({
+      Assets: moolaMint.mintPayment(moola(1)),
+    });
+    const { payout: payoutP, outcome: voterP, completeObj } = await E(
+      zoe,
+    ).offer(invite, proposal, payments);
+
+    const voter = await voterP;
+    const result = await E(voter).vote({ [QUESTION]: 'NO' });
+    t.equals(
+      result,
+      `Successfully voted ${{ [QUESTION]: 'NO' }}`,
+      `voter3 votes NOT`,
+    );
+
+    // Voter3 completes their offer and exits before the election is
+    // closed. Voter3's vote will not be counted.
+    completeObj.complete();
+
+    const payout = await payoutP;
+
+    const moolaPayment = await payout.Assets;
+
+    t.deepEquals(
+      await moolaIssuer.getAmountOf(moolaPayment),
+      moola(1),
+      `voter3 gets everything she escrowed back`,
+    );
+
+    console.log('EXPECTED ERROR ->>>');
+    t.throws(
+      () => voter.vote({ [QUESTION]: 'NO' }),
+      /the escrowing offer is no longer active/,
+      `voter3 voting fails once offer is withdrawn or amounts are reallocated`,
+    );
+  };
+
+  await voter3Votes(voterInvite3);
+
+  // Secretary closes election and tallies the votes.
+  const electionResults = await E(secretary).closeElection();
+  t.deepEquals(electionResults, { YES: 3, NO: 5 });
```

Comment:
It would be helpful to add another voter whose votes would be summed with one of the others.

Reply:
Yup, that is definitely needed

Reply:
Done